python_modules/dagster/dagster/daemon/sensor.py

```diff
@@ -175,6 +176,46 @@ def _check_for_debug_crash(debug_crash_flags, key):
 RELOAD_WORKSPACE = 60
 
 
+class SynchronousExecutor:
+    """
+    Executes functions in series without creating threads, creating a uniform execution interface
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        pass
+
+    def submit(self, fn, *args, **kwargs):
+        future = concurrent.futures.Future()
+
+        try:
+            result = fn(*args, **kwargs)
+            future.set_result(result)
+        except Exception as e:
+            future.set_exception(e)
+
+        return future
+
+    def shutdown(self, wait=True):
+        pass
+
+
+def sensor_executor(instance):
+    settings = instance.get_settings("sensors")
+
+    if settings.get("use_threads"):
+        return concurrent.futures.ThreadPoolExecutor(
+            max_workers=settings.get("num_workers"),
```

Comment:
See https://github.com/dagster-io/internal/pull/76 for some context on how we picked the default value for the cloud agent (not clear if that's totally applicable here - plus in the future we might have multiple threadpools for different daemons, we'll have to sort that out)